projects/js-packages/publicize-components/src/components/services/service-connection-info.tsx

```diff
@@ -32,15 +54,7 @@ export const ServiceConnectionInfo = ( { connection, service }: ServiceConnectio
 				{ connection.status === 'broken' ? (
 					<ConnectionStatus connection={ connection } service={ service } />
 				) : (
-					<div className={ styles[ 'mark-shared-wrap' ] }>
-						<MarkAsShared connection={ connection } />
-						<IconTooltip placement="top" inline={ false } shift>
-							{ __(
-								'If enabled, the connection will be available to all administrators, editors, and authors.',
-								'jetpack'
-							) }
-						</IconTooltip>
-					</div>
+					renderConnectionActions()
```

Comment:
IIFEs are the best when it comes to avoiding nested ternary operators.

```tsx
{ ( conn => {
	if ( conn.status === 'broken' ) {
		return <ConnectionStatus connection={ conn } service={ service } />;
	}

	if ( conn.can_disconnect ) {
		return (
			<div className={ styles[ 'mark-shared-wrap' ] }>
				<MarkAsShared connection={ conn } />
				<IconTooltip placement="top" inline={ false } shift>
					{ __(
						'If enabled, the connection will be available to all administrators, editors, and authors.',
						'jetpack'
					) }
				</IconTooltip>
			</div>
		);
	}

	return (
		<Text className={ styles.description }>
			{ __( 'This connection is added by a site administrator.', 'jetpack' ) }
		</Text>
	);
} )( connection ) }
```

Reply:
@manzoorwanijk done